In [ ]:
# Tugas 5: Statistik

In [ ]:
import pandas as pd
from IPython.display import display
df = pd.read_csv('Spotify-2000.csv', thousands=r',')

# Menunjukkan sampel data

In [ ]:
# Menunjukkan 5 baris data pertama
df.head(5)

Index                   Title             Artist            Top Genre  \
0      1                 Sunrise        Norah Jones      adult standards   
1      2             Black Night        Deep Purple           album rock   
2      3          Clint Eastwood           Gorillaz  alternative hip hop   
3      4           The Pretender       Foo Fighters    alternative metal   
4      5  Waitin' On A Sunny Day  Bruce Springsteen         classic rock   

   Year  Beats Per Minute (BPM)  Energy  Danceability  Loudness (dB)  \
0  2004                     157      30            53            -14   
1  2000                     135      79            50            -11   
2  2001                     168      69            66             -9   
3  2007                     173      96            43             -4   
4  2002                     106      82            58             -5   

   Liveness  Valence  Length (Duration)  Acousticness  Speechiness  Popularity  
0        11       68                201            94            3          71  
1        17       81                207            17            7          39  
2         7       52                341             2           17          69  
3         3       37                269             0            4          76  
4        10       87                256             1            3          59

In [ ]:
# Data-data lagu terbaru dan terlama
print('Maksimum:')
display(df.sort_values(['Year'], ascending=[0]).head(1))
print('Minimum:')
display(df.sort_values(['Year'], ascending=[1]).head(1))

Maksimum:


Index                                             Title        Artist  \
780    781  A Whiter Shade of Pale - Original Single Version  Procol Harum   

      Top Genre  Year  Beats Per Minute (BPM)  Energy  Danceability  \
780  album rock  2019                     150      66            25   

     Loudness (dB)  Liveness  Valence  Length (Duration)  Acousticness  \
780             -7         9       44                249            50   

     Speechiness  Popularity  
780            3          67

Minimum:


Index             Title         Artist        Top Genre  Year  \
1985   1986  Blue Suede Shoes  Elvis Presley  adult standards  1956   

      Beats Per Minute (BPM)  Energy  Danceability  Loudness (dB)  Liveness  \
1985                      95      66            56             -8        14   

      Valence  Length (Duration)  Acousticness  Speechiness  Popularity  
1985       96                119            65            6          62

Dari data di atas, terlihat bahwa data year memiliki range [1956, 2019].

In [ ]:
# Data-data dengan BPM tertinggi dan terendah
print('Maksimum:')
display(df.sort_values(['Beats Per Minute (BPM)'], ascending=[0]).head(1))
print('Minimum:')
display(df.sort_values(['Beats Per Minute (BPM)'], ascending=[1]).head(1))

Maksimum:


Index                                              Title  Artist  \
1426   1427  It's The End Of The World As We Know It (And I...  R.E.M.   

             Top Genre  Year  Beats Per Minute (BPM)  Energy  Danceability  \
1426  alternative rock  1987                     206      89            38   

      Loudness (dB)  Liveness  Valence  Length (Duration)  Acousticness  \
1426             -9         3       80                247             3   

      Speechiness  Popularity  
1426            4          67

Minimum:


Index                              Title      Artist     Top Genre  Year  \
998    999  Still Crazy After All These Years  Paul Simon  classic rock  1975   

     Beats Per Minute (BPM)  Energy  Danceability  Loudness (dB)  Liveness  \
998                      37      25            27            -12         9   

     Valence  Length (Duration)  Acousticness  Speechiness  Popularity  
998       13                207            80            4          61

Dari data di atas, terlihat bahwa data BPM memiliki range [37, 206].

In [ ]:
# Data-data dengan Energy tertinggi dan terendah
print('Maksimum:')
display(df.sort_values(['Energy'], ascending=[0]).head(1))
print('Minimum:')
display(df.sort_values(['Energy'], ascending=[1]).head(1))

Maksimum:


Index           Title  Artist          Top Genre  Year  \
1345   1346  Angel Of Death  Slayer  alternative metal  1986   

      Beats Per Minute (BPM)  Energy  Danceability  Loudness (dB)  Liveness  \
1345                      90     100            28             -4        15   

      Valence  Length (Duration)  Acousticness  Speechiness  Popularity  
1345       10                291             0           16          57

Minimum:


Index                                Title         Artist  \
1978   1979  The First Time Ever I Saw Your Face  Roberta Flack   

            Top Genre  Year  Beats Per Minute (BPM)  Energy  Danceability  \
1978  adult standards  1969                     121       3            31   

      Loudness (dB)  Liveness  Valence  Length (Duration)  Acousticness  \
1978            -22        16       14                261            73   

      Speechiness  Popularity  
1978            4          64

Dari data di atas, terlihat bahwa data Energy memiliki range [3, 100]

In [ ]:
# Data-data dengan Danceability tertinggi dan terendah
print('Maksimum:')
display(df.sort_values(['Danceability'], ascending=[0]).head(1))
display(df.sort_values(['Danceability'], ascending=[1]).head(1))

Maksimum:


Index          Title Artist      Top Genre  Year  \
1477   1478  Kingston Town   UB40  reggae fusion  1989   

      Beats Per Minute (BPM)  Energy  Danceability  Loudness (dB)  Liveness  \
1477                     102      25            96            -13         4   

      Valence  Length (Duration)  Acousticness  Speechiness  Popularity  
1477       80                228            16            6          49

Index                      Title Artist Top Genre  Year  \
1650   1651  Live Forever - Remastered  Oasis   britpop  1994   

      Beats Per Minute (BPM)  Energy  Danceability  Loudness (dB)  Liveness  \
1650                      84      79            10             -4        36   

      Valence  Length (Duration)  Acousticness  Speechiness  Popularity  
1650       18                277             0            4          55

Dari data di atas, terlihat bahwa data Danceability memiliki range [10, 96]

In [ ]:
# Data-data dengan Loudness tertinggi dan terendah
print('Maksimum:')
display(df.sort_values(['Loudness (dB)'], ascending=[0]).head(1))
print('Minimum:')
display(df.sort_values(['Loudness (dB)'], ascending=[1]).head(1))

Maksimum:


Index             Title   Artist Top Genre  Year  \
1818   1819  The Next Episode  Dr. Dre    g funk  1999   

      Beats Per Minute (BPM)  Energy  Danceability  Loudness (dB)  Liveness  \
1818                      95      91            92             -2         9   

      Valence  Length (Duration)  Acousticness  Speechiness  Popularity  
1818       31                162             3           27          70

Minimum:


Index Title           Artist  Top Genre  Year  Beats Per Minute (BPM)  \
1370   1371  Anne  Herman van Veen  dutch pop  1986                     115   

      Energy  Danceability  Loudness (dB)  Liveness  Valence  \
1370       8            67            -27        17       28   

      Length (Duration)  Acousticness  Speechiness  Popularity  
1370                228            67            4          42

Dari data di atas, terlihat bahwa data Loudness memiliki range [-27, -2] (dalam dB)

In [ ]:
# Data-data dengan Liveness tertinggi dan terendah
print('Maksimum:')
display(df.sort_values(['Liveness'], ascending=[0]).head(1))
print('Minimum:')
display(df.sort_values(['Liveness'], ascending=[1]).head(1))

Maksimum:


Index           Title           Artist       Top Genre  Year  \
405    406  Wonderful Days  Charly Lownoise  happy hardcore  2010   

     Beats Per Minute (BPM)  Energy  Danceability  Loudness (dB)  Liveness  \
405                     174      92            55            -11        99   

     Valence  Length (Duration)  Acousticness  Speechiness  Popularity  
405       11                332             1            7          37

Minimum:


Index   Title          Artist   Top Genre  Year  Beats Per Minute (BPM)  \
206    207  Bombay  Golden Earring  album rock  2008                     125   

     Energy  Danceability  Loudness (dB)  Liveness  Valence  \
206      92            52             -7         2       73   

     Length (Duration)  Acousticness  Speechiness  Popularity  
206                234             2            5          34

Dari data di atas, terlihat bahwa Liveness memiliki range [2, 99]

In [ ]:
# Data-data dengan Valence tertinggi dan terendah
print('Maksimum:')
display(df.sort_values(['Valence'], ascending=[0]).head(1))
print('Minimum:')
display(df.sort_values(['Valence'], ascending=[1]).head(1))

Maksimum:


Index                 Title               Artist   Top Genre  Year  \
1102   1103  What a Fool Believes  The Doobie Brothers  album rock  1978   

      Beats Per Minute (BPM)  Energy  Danceability  Loudness (dB)  Liveness  \
1102                     121      38            76            -15         5   

      Valence  Length (Duration)  Acousticness  Speechiness  Popularity  
1102       99                224            28            4          69

Minimum:


Index    Title     Artist   Top Genre  Year  Beats Per Minute (BPM)  \
1884   1885  The End  The Doors  album rock  1967                     106   

      Energy  Danceability  Loudness (dB)  Liveness  Valence  \
1884      30            35            -12        14        3   

      Length (Duration)  Acousticness  Speechiness  Popularity  
1884                703            16            3          61

Dari data di atas, terlihat bahwa Valence memiliki range [3, 99]

In [ ]:
# Data-data dengan Durasi tertinggi dan terendah
print('Maksimum:')
display(df.sort_values(['Length (Duration)'], ascending=[0]).head(1))
print('Minimum:')
display(df.sort_values(['Length (Duration)'], ascending=[1]).head(1))

Maksimum:


Index   Title      Artist   Top Genre  Year  Beats Per Minute (BPM)  \
842    843  Echoes  Pink Floyd  album rock  1971                     134   

     Energy  Danceability  Loudness (dB)  Liveness  Valence  \
842      32            28            -17        11       14   

     Length (Duration)  Acousticness  Speechiness  Popularity  
842               1412            37            4          58

Minimum:


Index          Title    Artist        Top Genre  Year  \
1049   1050  Stayin' Alive  Bee Gees  adult standards  1977   

      Beats Per Minute (BPM)  Energy  Danceability  Loudness (dB)  Liveness  \
1049                     104      54            71            -20         9   

      Valence  Length (Duration)  Acousticness  Speechiness  Popularity  
1049       64                 93            11            4          60

Dari data di atas, terlihat bahwa Duration memiliki range [102, 1412] (dalam detik).

In [ ]:
# Data-data dengan Acousticness tertinggi dan terendah
print('Maksimum:')
display(df.sort_values(['Acousticness'], ascending=[0]).head(1))
print('Minimum:')
display(df.sort_values(['Acousticness'], ascending=[1]).head(1))

Maksimum:


Index                                  Title        Artist Top Genre  \
170    171  Comptine d'un autre été, l'après-midi  Yann Tiersen   bow pop   

     Year  Beats Per Minute (BPM)  Energy  Danceability  Loudness (dB)  \
170  2001                     100      19            48            -15   

     Liveness  Valence  Length (Duration)  Acousticness  Speechiness  \
170        11        6                141            99            5   

     Popularity  
170          56

Minimum:


Index                    Title       Artist   Top Genre  Year  \
1840   1841  House Of The Rising Sun  The Animals  album rock  1964   

      Beats Per Minute (BPM)  Energy  Danceability  Loudness (dB)  Liveness  \
1840                     117      58            30             -7        10   

      Valence  Length (Duration)  Acousticness  Speechiness  Popularity  
1840       23                271             0            3          69

Dari data di atas, terlihat bahwa Acousticness memiliki range [0, 99]

In [ ]:
# Data-data dengan Speechiness tertinggi dan terendah
display(df.sort_values(['Speechiness'], ascending=[0]).head(1))
print('Minimum:')
display(df.sort_values(['Speechiness'], ascending=[1]).head(1))

Index          Title        Artist   Top Genre  Year  \
864    865  Mercedes Benz  Janis Joplin  blues rock  1971   

     Beats Per Minute (BPM)  Energy  Danceability  Loudness (dB)  Liveness  \
864                      87      11            70            -15         7   

     Valence  Length (Duration)  Acousticness  Speechiness  Popularity  
864       52                108            94           55          58

Minimum:


Index                       Title     Artist   Top Genre  Year  \
1063   1064  Two Out of Three Ain't Bad  Meat Loaf  album rock  1977   

      Beats Per Minute (BPM)  Energy  Danceability  Loudness (dB)  Liveness  \
1063                      80      48            59             -9         9   

      Valence  Length (Duration)  Acousticness  Speechiness  Popularity  
1063       49                324            39            2          60

Dari data di atas, terlihat bahwa Speechiness memiliki range [2, 55]

In [ ]:
# Data-data dengan Popularity tertinggi dan terendah
display(df.sort_values(['Popularity'], ascending=[0]).head(1))
print('Minimum:')
display(df.sort_values(['Popularity'], ascending=[1]).head(1))

Index         Title       Artist       Top Genre  Year  \
793    794  Dance Monkey  Tones and I  australian pop  2019   

     Beats Per Minute (BPM)  Energy  Danceability  Loudness (dB)  Liveness  \
793                      98      59            82             -6        15   

     Valence  Length (Duration)  Acousticness  Speechiness  Popularity  
793       51                209            69            9         100

Minimum:


Index      Title Artist    Top Genre  Year  Beats Per Minute (BPM)  \
1793   1794  Sacrifice  Anouk  dutch indie  1998                     136   

      Energy  Danceability  Loudness (dB)  Liveness  Valence  \
1793      17            54            -13         9       24   

      Length (Duration)  Acousticness  Speechiness  Popularity  
1793                238            74            5          11

Dari data di atas, terlihat bahwa Popularity memiliki range [11, 100]

# Statistik rata-rata dan standar deviasi

In [ ]:
# Statistik rata-rata data dan standar deviasi.
print('Rata-rata:')
display(df[['Beats Per Minute (BPM)', 'Energy', 
   'Danceability', 'Loudness (dB)', 'Liveness', 'Valence', 'Length (Duration)',
   'Acousticness', 'Speechiness', 'Popularity']].mean())
print('Standar deviasi:')
display(df[['Beats Per Minute (BPM)', 'Energy', 
   'Danceability', 'Loudness (dB)', 'Liveness', 'Valence', 'Length (Duration)',
   'Acousticness', 'Speechiness', 'Popularity']].std())

Rata-rata:


Beats Per Minute (BPM)    120.215647
Energy                     59.679539
Danceability               53.238215
Loudness (dB)              -9.008526
Liveness                   19.012036
Valence                    49.408726
Length (Duration)         262.443330
Acousticness               28.858074
Speechiness                 4.994985
Popularity                 59.526580
dtype: float64

Standar deviasi:


Beats Per Minute (BPM)    28.028096
Energy                    22.154322
Danceability              15.351507
Loudness (dB)              3.647876
Liveness                  16.727378
Valence                   24.858212
Length (Duration)         93.604387
Acousticness              29.011986
Speechiness                4.401566
Popularity                14.351600
dtype: float64

# Statistik percentile

In [ ]:
# Statistik percentile (10%, 25%, 50%, 75%, dan 90%).
print('Persentil:')
display(df[['Beats Per Minute (BPM)', 'Energy', 
   'Danceability', 'Loudness (dB)', 'Liveness', 'Valence', 'Length (Duration)',
   'Acousticness', 'Speechiness', 'Popularity']].quantile([0.10, 0.25, 0.50, 0.75, 0.90]))

Persentil:


Beats Per Minute (BPM)  Energy  Danceability  Loudness (dB)  Liveness  \
0.10                    84.0    29.0          32.0          -14.0       7.0   
0.25                    99.0    42.0          43.0          -11.0       9.0   
0.50                   119.0    61.0          53.0           -8.0      12.0   
0.75                   136.0    78.0          64.0           -6.0      23.0   
0.90                   162.0    89.0          73.0           -5.0      37.0   

      Valence  Length (Duration)  Acousticness  Speechiness  Popularity  
0.10    18.00              181.3           0.0          3.0       40.00  
0.25    29.00              212.0           3.0          3.0       49.25  
0.50    47.00              245.0          18.0          4.0       62.00  
0.75    69.75              289.0          50.0          5.0       71.00  
0.90    86.00              349.7          75.0          8.0       76.00

# Statistik ekstremum (nilai maksimum dan minimum)

In [ ]:
# Statistik ekstremum maksimum dan minimum
print('Maksimum: ')
display(df[['Year', 'Beats Per Minute (BPM)', 'Energy', 
   'Danceability', 'Loudness (dB)', 'Liveness', 'Valence', 'Length (Duration)',
   'Acousticness', 'Speechiness', 'Popularity']].max())
print('Minimum: ')
display(df[['Year', 'Beats Per Minute (BPM)', 'Energy', 
   'Danceability', 'Loudness (dB)', 'Liveness', 'Valence', 'Length (Duration)',
   'Acousticness', 'Speechiness', 'Popularity']].min())

Maksimum: 


Year                      2019
Beats Per Minute (BPM)     206
Energy                     100
Danceability                96
Loudness (dB)               -2
Liveness                    99
Valence                     99
Length (Duration)         1412
Acousticness                99
Speechiness                 55
Popularity                 100
dtype: int64

Minimum: 


Year                      1956
Beats Per Minute (BPM)      37
Energy                       3
Danceability                10
Loudness (dB)              -27
Liveness                     2
Valence                      3
Length (Duration)           93
Acousticness                 0
Speechiness                  2
Popularity                  11
dtype: int64

In [ ]:
# Statistik untuk setiap genre.
print('Mean:')
display(df.groupby('Top Genre')[['Beats Per Minute (BPM)', 'Energy', 
   'Danceability', 'Loudness (dB)', 'Liveness', 'Valence',
   'Acousticness', 'Speechiness', 'Popularity']].mean().dropna())
print('Standar Deviasi:')
display(df.groupby('Top Genre')[['Beats Per Minute (BPM)', 'Energy', 
   'Danceability', 'Loudness (dB)', 'Liveness', 'Valence',
   'Acousticness', 'Speechiness', 'Popularity']].std().dropna())
print('Nilai Maksimum:')
display(df.groupby('Top Genre')[['Beats Per Minute (BPM)', 'Energy', 
   'Danceability', 'Loudness (dB)', 'Liveness', 'Valence',
   'Acousticness', 'Speechiness', 'Popularity']].max().dropna())
print('Nilai Minimum:')
display(df.groupby('Top Genre')[['Beats Per Minute (BPM)', 'Energy', 
   'Danceability', 'Loudness (dB)', 'Liveness', 'Valence',
   'Acousticness', 'Speechiness', 'Popularity']].min().dropna())

Mean:


Beats Per Minute (BPM)     Energy  Danceability  \
Top Genre                                                           
acid jazz                      92.000000  95.000000     66.000000   
acoustic pop                  123.750000  28.250000     48.750000   
adult standards               117.105691  42.658537     49.162602   
afropop                       154.000000  70.000000     68.000000   
alaska indie                  131.000000  22.000000     46.000000   
...                                  ...        ...           ...   
stomp and holler              102.000000  73.000000     48.000000   
streektaal                     98.000000  21.000000     61.000000   
trance                        138.000000  85.000000     60.000000   
uk pop                         82.000000  25.000000     53.000000   
yacht rock                    129.000000  92.000000     74.000000   

                  Loudness (dB)   Liveness    Valence  Acousticness  \
Top Genre                                                             
acid jazz             -5.000000  14.000000  52.000000     13.000000   
acoustic pop         -11.250000  16.500000  25.750000     58.250000   
adult standards      -11.504065  19.243902  53.398374     51.382114   
afropop              -10.000000  33.000000  51.000000      8.000000   
alaska indie         -10.000000  11.000000  15.000000     90.000000   
...                         ...        ...        ...           ...   
stomp and holler      -5.000000  21.000000  43.000000      3.000000   
streektaal           -20.000000   8.000000  42.000000     60.000000   
trance                -6.000000   6.000000   9.000000      1.000000   
uk pop               -16.000000  12.000000  34.000000     52.000000   
yacht rock            -5.000000   8.000000  93.000000     11.000000   

                  Speechiness  Popularity  
Top Genre                                  
acid jazz           11.000000   69.000000  
acoustic pop         3.750000   66.750000  
adult standards      3.926829   62.577236  
afropop              3.000000   59.000000  
alaska indie         3.000000   41.000000  
...                       ...         ...  
stomp and holler     3.000000   64.000000  
streektaal           3.000000   28.000000  
trance               6.000000   58.000000  
uk pop               3.000000   61.000000  
yacht rock           4.000000   64.000000  

[149 rows x 9 columns]

Standar Deviasi:


Beats Per Minute (BPM)     Energy  Danceability  \
Top Genre                                                                     
acoustic pop                             34.052655  13.073510     18.080837   
adult standards                          28.898866  19.874949     16.363381   
album rock                               26.011898  20.794664     14.451125   
alternative dance                        14.065493  10.140912     14.226735   
alternative hip hop                      20.506097   1.414214     11.313708   
...                                            ...        ...           ...   
pop                                      25.555910  24.215900     16.931707   
punk                                      9.899495   4.949747      3.535534   
reggae                                   24.927036  17.547894     10.920982   
reggae fusion                            37.674041  15.438048      8.180261   
scottish singer-songwriter               11.313708   0.707107      1.414214   

                            Loudness (dB)   Liveness    Valence  Acousticness  \
Top Genre                                                                       
acoustic pop                     2.753785  11.733144  30.966380     32.857013   
adult standards                  3.722713  17.426822  26.976535     30.287562   
album rock                       3.582561  14.938927  24.112347     25.627058   
alternative dance                2.077086  13.595517  14.588972      3.943651   
alternative hip hop              1.414214  38.183766  17.677670      0.707107   
...                                   ...        ...        ...           ...   
pop                              3.744376  13.239696  26.563958     28.459686   
punk                             0.707107   3.535534  21.920310      2.828427   
reggae                           1.246423  36.699552  11.230539     19.639610   
reggae fusion                    2.000000  10.082989   6.806859      7.410578   
scottish singer-songwriter       0.000000   2.828427  29.698485     16.970563   

                            Speechiness  Popularity  
Top Genre                                            
acoustic pop                   0.957427   12.526638  
adult standards                1.878134    9.824700  
album rock                     4.200491   12.880100  
alternative dance              1.183216   11.564725  
alternative hip hop            0.707107    7.778175  
...                                 ...         ...  
pop                            2.854634   14.534147  
punk                           3.535534    2.121320  
reggae                        11.625096   12.391673  
reggae fusion                  1.290994   12.124356  
scottish singer-songwriter     0.000000    7.778175  

[95 rows x 9 columns]

Nilai Maksimum:


Beats Per Minute (BPM)  Energy  Danceability  Loudness (dB)  \
Top Genre                                                                       
acid jazz                             92      95            66             -5   
acoustic pop                         151      46            69             -8   
adult standards                      184      97            88             -5   
afropop                              154      70            68            -10   
alaska indie                         131      22            46            -10   
...                                  ...     ...           ...            ...   
stomp and holler                     102      73            48             -5   
streektaal                            98      21            61            -20   
trance                               138      85            60             -6   
uk pop                                82      25            53            -16   
yacht rock                           129      92            74             -5   

                  Liveness  Valence  Acousticness  Speechiness  Popularity  
Top Genre                                                                   
acid jazz               14       52            13           11          69  
acoustic pop            34       72            91            5          82  
adult standards         96       97            97           17          78  
afropop                 33       51             8            3          59  
alaska indie            11       15            90            3          41  
...                    ...      ...           ...          ...         ...  
stomp and holler        21       43             3            3          64  
streektaal               8       42            60            3          28  
trance                   6        9             1            6          58  
uk pop                  12       34            52            3          61  
yacht rock               8       93            11            4          64  

[149 rows x 9 columns]

Nilai Minimum:


Beats Per Minute (BPM)  Energy  Danceability  Loudness (dB)  \
Top Genre                                                                       
acid jazz                             92      95            66             -5   
acoustic pop                          74      17            33            -14   
adult standards                       61       3            16            -22   
afropop                              154      70            68            -10   
alaska indie                         131      22            46            -10   
...                                  ...     ...           ...            ...   
stomp and holler                     102      73            48             -5   
streektaal                            98      21            61            -20   
trance                               138      85            60             -6   
uk pop                                82      25            53            -16   
yacht rock                           129      92            74             -5   

                  Liveness  Valence  Acousticness  Speechiness  Popularity  
Top Genre                                                                   
acid jazz               14       52            13           11          69  
acoustic pop             9        7            13            3          56  
adult standards          4        6             0            2          31  
afropop                 33       51             8            3          59  
alaska indie            11       15            90            3          41  
...                    ...      ...           ...          ...         ...  
stomp and holler        21       43             3            3          64  
streektaal               8       42            60            3          28  
trance                   6        9             1            6          58  
uk pop                  12       34            52            3          61  
yacht rock               8       93            11            4          64  

[149 rows x 9 columns]

# Statistik per genre lagu

In [ ]:
# Mencari informasi-informasi yang menarik per genre.
display(df.groupby('Top Genre')[['Beats Per Minute (BPM)', 'Energy', 
   'Danceability', 'Loudness (dB)', 'Liveness', 'Valence',
   'Acousticness', 'Speechiness', 'Popularity']].mean().dropna().sort_values('Beats Per Minute (BPM)'))
print('Rata-rata menunjukkan genre indie anthem-folk memiliki BPM paling rendah dan genre laboratorio memiliki BPM paling tinggi.')
display(df.groupby('Top Genre')[['Beats Per Minute (BPM)', 'Energy', 
   'Danceability', 'Loudness (dB)', 'Liveness', 'Valence',
   'Acousticness', 'Speechiness', 'Popularity']].mean().dropna().sort_values('Popularity'))
print('Rata-rata menunjukkan genre classic italian pop memiliki Popularity paling rendah dan genre indie pop dan celtic punk memiliki Popularity paling tinggi.')

Beats Per Minute (BPM)  Energy  Danceability  \
Top Genre                                                                 
indie anthem-folk                            69.0    62.0          59.0   
cyberpunk                                    74.0    30.0          43.0   
contemporary country                         78.5    54.5          48.5   
british singer-songwriter                    79.0    60.0          51.0   
atl hip hop                                  80.0    97.0          73.0   
...                                           ...     ...           ...   
bebop                                       174.0    26.0          45.0   
latin alternative                           176.0    74.0          66.0   
latin                                       178.5    67.0          55.5   
electro house                               180.0    63.0          47.0   
laboratorio                                 189.0    76.0          43.0   

                           Loudness (dB)  Liveness  Valence  Acousticness  \
Top Genre                                                                   
indie anthem-folk                   -6.0      13.0     27.0           2.0   
cyberpunk                          -14.0      23.0     16.0          70.0   
contemporary country                -7.5      16.0     41.5          12.0   
british singer-songwriter           -9.0      11.0     40.0          44.0   
atl hip hop                         -2.0      18.0     97.0          10.0   
...                                  ...       ...      ...           ...   
bebop                              -13.0       7.0     60.0          54.0   
latin alternative                   -7.0      22.0     91.0          66.0   
latin                               -6.0       6.5     72.5          44.0   
electro house                       -5.0      11.0     16.0           3.0   
laboratorio                         -8.0      55.0     66.0           2.0   

                           Speechiness  Popularity  
Top Genre                                           
indie anthem-folk                 11.0        75.0  
cyberpunk                          3.0        61.0  
contemporary country               3.0        65.5  
british singer-songwriter          3.0        66.0  
atl hip hop                        7.0        79.0  
...                                ...         ...  
bebop                              4.0        65.0  
latin alternative                  8.0        70.0  
latin                              9.0        71.0  
electro house                      5.0        80.0  
laboratorio                        4.0        57.0  

[149 rows x 9 columns]

Rata-rata menunjukkan genre indie anthem-folk memiliki BPM paling rendah dan genre laboratorio memiliki BPM paling tinggi.


Beats Per Minute (BPM)  Energy  Danceability  \
Top Genre                                                                
classic italian pop                        150.0    40.0          45.0   
compositional ambient                      105.0    24.0          61.0   
streektaal                                  98.0    21.0          61.0   
british alternative rock                    87.5    65.5          54.5   
dutch prog                                 130.0    55.6          50.6   
...                                          ...     ...           ...   
electro house                              180.0    63.0          47.0   
ccm                                        148.0    63.0          49.0   
australian psych                           121.0    81.0          62.0   
indie pop                                  128.0    79.0          65.5   
celtic punk                                119.0    65.0          32.0   

                          Loudness (dB)  Liveness  Valence  Acousticness  \
Top Genre                                                                  
classic italian pop               -12.0       6.0     17.0           3.0   
compositional ambient             -17.0      25.0     37.0          76.0   
streektaal                        -20.0       8.0     42.0          60.0   
british alternative rock           -7.5      21.0     38.0           5.0   
dutch prog                         -9.6      22.4     45.6           4.4   
...                                 ...       ...      ...           ...   
electro house                      -5.0      11.0     16.0           3.0   
ccm                                -7.0       9.0      8.0          68.0   
australian psych                   -5.0      14.0     68.5           0.5   
indie pop                          -5.0      18.0     79.5           7.5   
celtic punk                        -6.0      24.0     47.0          50.0   

                          Speechiness  Popularity  
Top Genre                                          
classic italian pop               3.0        24.0  
compositional ambient             3.0        28.0  
streektaal                        3.0        28.0  
british alternative rock          3.0        28.0  
dutch prog                        3.8        31.2  
...                               ...         ...  
electro house                     5.0        80.0  
ccm                               3.0        80.0  
australian psych                  3.5        80.5  
indie pop                         4.5        83.0  
celtic punk                       3.0        83.0  

[149 rows x 9 columns]

Rata-rata menunjukkan genre classic italian pop memiliki Popularity paling rendah dan genre indie pop dan celtic punk memiliki Popularity paling tinggi.


# Distribusi frekuensi top genre pada data

In [ ]:
# Distribusi frekuensi genre pada data.
display(df['Top Genre'].value_counts())

album rock          413
adult standards     123
dutch pop            88
alternative rock     86
dance pop            83
                   ... 
hard rock             1
pop punk              1
alaska indie          1
gangster rap          1
laboratorio           1
Name: Top Genre, Length: 149, dtype: int64

Dari data di atas, dapat disimpulkan bahwa genre lagu yang paling banyak muncul dari tahun 1956 sampai dengan 2019 adalah album rock.